### PR TITLE
Fixes in /admin/groups route and admin/user route fixes( user role update )

### PR DESCRIPTION
**Pull request by kimenyikevin**
## Summary

_A summary of changes being made in this PR_

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

_References to pertaining issue(s)_

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_

## Task to be complete 
 - Fix update user role manually refresh
 - Fix update user role for your own account 
 - Refactor group /route 
 - Fix grate user access from group /route 
 - Fix search group functionality

### CODE REVIEW

**Comment by zulqarnainhanif** [comment]
@kimenyikevin admin user is unable to update role of other users as well now, need to fix that.

**Comment by zulqarnainhanif** [comment]
@kimenyikevin  pagination count of groups table is wrong

Screenshot 2022-02-03 at 12 01 56 PM

**Comment by zulqarnainhanif** [comment]
@kimenyikevin layout font sizes and spacing needs to be adjusted for View group

Screenshot 2022-02-03 at 12 01 06 PM

**Comment by zulqarnainhanif** [comment]
@HexaField how do we add more users in a group?

File: packages/client-core/src/admin/components/Group/CreateGroup.tsx [hunk 1]
```diff
@@ -80,13 +70,24 @@ const CreateGroup = (props: Props) => {
         description: '',
         scopeTypes: []
       })
+      handleClose(false)
     }
-    handleClose(false)
+  }
+
+  const scopeData: ScopeData[] = []
+  adminScopeTypeState.scopeTypes.value.forEach((el) => {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Group/CreateGroup.tsx:78]: you can use `map` here to declare and initialise the scopeData at the same time.

avoid usage of forEach as much as possible it use callback returns in the backend and doesn't stop execution.

File: packages/client-core/src/admin/components/Group/EditGroup.tsx [hunk 2]
```diff
@@ -87,6 +74,16 @@ const EditGroup = (props: Props) => {
       if (typeof closeViewModal === 'function') closeViewModal()
     }
   }
+  const handleChangeScopeType = (scope) => {
+    setState({ ...state, scopeTypes: scope, formErrors: { ...state.formErrors, scopeTypes: '' } })
+  }
+
+  const scopeData: ScopeData[] = []
+  adminScopeTypeState.scopeTypes.value.forEach((el) => {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Group/EditGroup.tsx:82]: same as above

File: packages/client-core/src/admin/components/Users/CreateUser.tsx [hunk 3]
```diff
@@ -184,81 +154,49 @@ const CreateUser = (props: Props) => {
     })
   })
 
+  const staticResourceMenu: InputSelectProps[] = []
+  staticResourceData.value.forEach((el) => {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Users/CreateUser.tsx:158]: same as above

File: packages/client-core/src/admin/components/Users/CreateUser.tsx [hunk 4]
```diff
@@ -184,81 +154,49 @@ const CreateUser = (props: Props) => {
...
+    staticResourceMenu.push({
+      label: el.name,
+      value: el.name
+    })
+  })
+
+  const userRoleData: InputSelectProps[] = []
+  userRole.userRole.value.forEach((el) => {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Users/CreateUser.tsx:166]: same as above

File: packages/client-core/src/admin/components/Users/ViewUser.tsx [hunk 5]
```diff
@@ -174,6 +185,14 @@ const ViewUser = (props: Props) => {
     })
   })
 
+  const userRoleData: InputSelectProps[] = []
+  userRole.userRole.value.forEach((el) => {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Users/ViewUser.tsx:189]: same as above

**Comment by zulqarnainhanif** [comment]
@kimenyikevin we are able to add same scope multiple times in the AutoComplete Component, fix that where ever AutoComplete Component is used. (Either disable selected one in the list or unselect them if already selected)

Screenshot 2022-02-03 at 9 37 17 PM

Screenshot 2022-02-03 at 9 36 49 PM

**Comment by HexaField** [comment]
> @HexaField how do we add more users in a group?

they need to be invited

**Comment by zulqarnainhanif** [comment]
@kimenyikevin 1 small use-case

Click view user
Click edit button
Edit some scopes from autocomplete
Press Cancel
Press Edit again 

Problem: The values are no reset in this case as the cancel should reset fields. This is working fine with groups.

**Review by zulqarnainhanif — APPROVED** [review]

**Pull request merged**